scripts/changelog.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable import/prefer-default-export */
+
+const assert = require('assert');
+const semver = require('semver');
+const fs = require('fs');
+const path = require('path');
+const marked = require('marked');
+const moment = require('moment');
+
+// TODO stronger check for version heading?
+const getPreviousVersion = (tokens) =>
+  tokens.find((t) => /v\d+/.test(t.text)).tokens.find((t) => t.type === 'text')
+    .text;
+
+const releaseHeading = (version) =>
+  `## <sub>${version}</sub>
+
+#### ${moment().format('_MMM. D, YYYY_')}\n\n`;
+
+const pendingReleaseNotes = (changelog) => {
+  const tokens = marked.lexer(changelog.toString());
+
+  const previousFullReleaseIndex = tokens.findIndex(
+    (t) => t.type === 'heading' && t.depth === 2 && !/alpha/.test(t.text)
+  );
+
+  return tokens.slice(0, previousFullReleaseIndex);
+};
+
+const pastReleaseNotes = (changelog) => {
+  const tokens = marked.lexer(changelog.toString());
+
+  const previousFullReleaseIndex = tokens.findIndex(
+    (t) => t.type === 'heading' && t.depth === 2 && !/alpha/.test(t.text)
+  );
+
+  return tokens.slice(previousFullReleaseIndex);
+};
+
+const isPendingPreRelease = (tokens) => tokens[0].type !== 'heading';
+
+/**
+ * Given a list of tokens and a heading, find all list items that appear
+ * under that heading.
+ */
+const getItemsFor = (heading, tokens) => {
+  let items = '';
+
+  let i = 0;
+
+  while (i < tokens.length) {
+    const currentToken = tokens[i];
+    const regex = new RegExp(heading, 'i');
+
+    if (currentToken.type === 'paragraph' && regex.test(currentToken.text)) {
+      // search for list
+      let j = i + 1;
+      while (j < tokens.length) {
+        if (tokens[j].type === 'list') {
+          items += `${tokens[j].raw.trim()}\n`;
+          i = j + 1;
+          break;
+        } else {
+          j += 1;
+        }
+      }
+    } else {
+      i += 1;
+    }
+  }
+
+  return items.length ? `${items}\n` : '';
+};
+
+const findSubHeadings = (tokens) => {
+  const headings = new Set();
+  const regex = /\*\*([a-zA-Z ]+)\*\*/;
```

Comment:
I noticed that sometimes we use:
`**New**`
and sometimes:
`**New:**`

Reply:
and also sometimes `**New**:` - I'm thinking a lint tool for the changelog might be a better option - thoughts?

Reply:
Yeah good shout.  If we are deciding here I prefer the colon-less version 🍰

Reply:
yeah, I think so too. Could also change it to a heading instead of some bold text - but no need to change too much. I'll look into how easy it is to write lint rules and/or prettier extension.